variable/include/scipp/variable/variable.h

```diff
@@ -366,6 +366,18 @@ SCIPP_VARIABLE_EXPORT VariableView copy(const VariableConstView &dataset,
 
 } // namespace scipp::variable
 
+namespace scipp::core {
+template <> constexpr DType dtype<variable::Variable>{1000};
+template <>
+constexpr DType dtype<variable::VariableView>{
+    1001}; // hack for python bindings using
+           // dtype<ElementArrayView<bucket<Variable>>::value_type>
+           // is setting same dtype ID correct?
+template <> constexpr DType dtype<bucket<variable::Variable>>{1001};
+template <> constexpr DType dtype<bucket<variable::VariableConstView>>{1002};
+template <> constexpr DType dtype<bucket<variable::VariableView>>{1003};
+} // namespace scipp::core
```

Comment:
Already commented on, but since these index assignements are split across the code base, makes me think more that a better solution for reasonable to identify clashes at cmake/compile time needs to be found as some follow up.

Reply:
Well, that is impossible in general I suppose, since we aim at publishing our package, so anyone can instantiate a new variable. Of course we can ensure it works within our own libraries, somehow.